nixos/modules/services/databases/redis.nix

```diff
@@ -277,11 +283,18 @@ in
 
       serviceConfig = {
         ExecStart = "${cfg.package}/bin/redis-server /run/redis/redis.conf";
-        RuntimeDirectory = "redis";
-        StateDirectory = "redis";
         Type = "notify";
+        # User and group
         User = "redis";
         Group = "redis";
+        # Runtime directory and mode
+        RuntimeDirectory = "redis";
+        RuntimeDirectoryMode = "0750";
+        # State directory and mode
+        StateDirectory = "redis";
+        StateDirectoryMode = "0750";
+        # Access write directories
+        UMask = "0027";
```

Comment:
What is this needed for?

Reply:
Need to simple configure service `mastodon` and other services that can use the unix socket:
```
      (lib.mkIf cfg.redis.enableUnixSocket {
        unixSocket = "/run/redis/redis.sock";
        unixSocketPerm = 770;
      })
```

Reply:
Need to enable access to `dump.rdb` file only redis group.
```
sudo ls -lah /var/lib/redis
итого 557K
drwxr-x---  2 redis redis    3 мар 31 15:37 .
drwxr-xr-x 15 root  root  4,0K мар 31 15:05 ..
-rw-r-----  1 redis redis 693K мар 31 15:37 dump.rdb

```

Reply:
Also here, why does redis-clients need to access `/var/lib/redis`?

Reply:
Changed to `UMask = "0077";`